File: ios/Runner/BeaconSdk/Disgnostic/SdkDiagnostic.swift
import Foundation

struct SdkDiagnostic {
  let state: String
  let isInitialized: Bool
  let isMonitoring: Bool
  let lastStep: String?
  let lastErrorCode: Int?
  let lastErrorMessage: String?
  let timestamp: Int64
  
  init(state: String, 
      isInitialized: Bool, 
      isMonitoring: Bool, 
      lastStep: String?, 
      lastErrorCode: Int?, 
      lastErrorMessage: String?) {
    
    self.state = state
    self.isInitialized = isInitialized
    self.isMonitoring = isMonitoring
    self.lastStep = lastStep
    self.lastErrorCode = lastErrorCode
    self.lastErrorMessage = lastErrorMessage
    self.timestamp = Int64(Date().timeIntervalSince1970 * 1000)
  }
}

File: android/beaconsdk/src/main/java/com/xenber/beaconsdk/diagnostic/SdkDiagnostic.kt
package com.xenber.beaconsdk.diagnostic

data class SdkDiagnostic(
    val state: String,
    val isInitialized: Boolean,
    val isMonitoring: Boolean,
    val lastStep: String?,
    val lastErrorCode: Int?,
    val lastErrorMessage: String?,
    val timestamp: Long = System.currentTimeMillis()
)